csrc/mutator.cpp

```diff
@@ -107,13 +107,30 @@ void OptOutMutator::mutate(TensorDomain* td) {
     return;
   }
 
+  // Update the contiguity vector. Drop the contig val if mutated to broadcast
+  auto contig = td->contiguity();
+
+  for (const auto i : c10::irange(td->maybeRFactor().size())) {
+    auto original_id = td->maybeRFactor().at(i);
+    if (original_id->getIterType() != IterType::Symbolic) {
+      continue;
+    }
+
+    TORCH_INTERNAL_ASSERT(
+        contig.at(i),
+        "Unexpected to have a non-contig symbolic domain: ",
+        original_id->toString());
+
+    auto updated_id = td->hasRFactor() ? rfactor_dom.at(i) : root_dom.at(i);
+
+    // If the mutated ID is a broadcast domain, drop the contig val
+    if (updated_id->isBroadcast()) {
+      contig.at(i) = std::nullopt;
+    }
+  }
```

Comment:
This was previously done in `dynamic_transform.cpp` but I think it makes sense to always recompute `contig` if mutating to `Broadcast`.

Reply:
Note that this only covers the case where the original ID was `Symbolic` and checks that it was marked contig then. If we mutated from an original ID with type `Broadcast` or `Iteration` we might want to fill in a different contiguity here instead.